src/ssort/_requirements.py

```diff
@@ -44,6 +52,30 @@ def _get_scope_from_arguments(args: ast.arguments) -> set[str]:
     return scope
 
 
+if sys.version_info >= (3, 12):
+
+    def _get_scope_from_type_params(
+        type_params: list[ast.type_param],
+    ) -> set[str]:
+        return set(type_param.name for type_param in type_params)  # type: ignore[attr-defined]
+
+    @get_requirements.register(ast.TypeAlias)
+    def _get_requirements_for_type_alias(
+        node: ast.TypeAlias,
+    ) -> Iterable[Requirement]:
+        scope = _get_scope_from_type_params(node.type_params)
+        for requirement in _get_requirements_from_nodes(node.type_params):
+            if requirement.name not in scope:
+                yield requirement
```

Comment:
It's now using the scope from type params to filter their own requirements. This adds the compliance to:

> Type parameters are visible to other type parameters declared elsewhere in the list. This allows type parameters to use other type parameters within their definition. 

for the PEP. It's not considering any order for sake of simplicity, leaving the more strict rules to the compiler or type checkers.

Reply:
Yup, had read that as scope being updated from left to right but I think you are right.